crates/iota-faucet/src/faucet/simple_faucet.rs

```diff
@@ -131,16 +136,19 @@ impl SimpleFaucet {
             config.max_request_queue_length as usize,
         );
 
-        // This is to handle the case where there is only 1 coin, we want it to go to
-        // the normal queue
-        let split_point = if coins.len() > 10 {
-            coins.len() / 2
```

Comment:
@samuel-rufi could you elaborate on how the gas pool handles gas tokens? Why were there multiple gas coins put into it before and why do we only put one now? Is the faucet capable of issuing concurrent transactions with different gas coins?

Reply:
Great question @lzpap !

So first of all each user request to the gas pool will be handled in a dedicated task spawn in https://github.com/iotaledger/iota/blob/a403cf304ba14b5b5cfbbc8a60a78e00e7bb5e4b/crates/iota-faucet/src/server.rs#L205.

Each task, will try to get a free gas object from the queue (protected by an Mutex to make sure each request gets a different gas object). The gas pool can serve multiple requests using multiple gas objects basically at the same time.

So how does it work in more detail? For each TX, it will try to get an available coin from this queue. This coin will be locked and unlocked (also called "recycled") once the transaction was successful. If the transaction fails, the coin will stay locked and the queue with available objects shrinks.

So in the case `batch` is enabled, we only allocate only one gas object to the normal gas pool and send all other to the batch pool. In case `batch` is disabled, all gas coins will be added to the gas pool.

In case we don't want to use batch mode, but still want to be able to serve multiple faucet requests (basically at the same time) we should provide multiple gas objects to the gas pool.

Reply:
Thanks for the answers!
So currently on the testnet we have the non-batch mode and 2 objects in the gas pool ([obj 1](https://explorer.rebased.iota.org/object/0x52efb746f32d44486e581f7f166ebadc179278422ece1e1c74dcbb04832ef3d6?network=testnet)) and ([obj 2](https://explorer.rebased.iota.org/object/0x94bfb7d5b8a977b8a168eb74b5515fe132b190847efa332b3df8f361291cf8fc?network=testnet)), right?
@junwei0117 next time the faucet gets topped up would be nice to send the coins in smaller batches to it, so it has 5-10 coins and can serve concurrent requests as well.

Reply:
In case a request comes in and there is no gas coin available (for both modes), it will wait some seconds and then return an error if the queue is still empty.